### DISCUSSION TITLE
关于不同链与链合约的适配问题

### DISCUSSION THREAD

**Discussion by 0xKT** [NonceGeek/Tai-Shang-Soul-Card#130] [Ideas]
问题：目前采用的是，单一适配，只做了V链的合约接口配置，如果切换了链，但其实内部的接口还是原来链的合约配置，因此强行调用，会出现错误

**Comment by 0xKT** [comment]
解决方法
1. 部署所有支持链的合约
2. 新建一个索引表，以ChainID->组织->合约地址,ChainID->个人->合约地址，这是思考是在后端做，还是前端绑定
3. 根据以上信息，重构合约调用接口

**Reply by 0xKT** [reply]
或者name->合约地址

**Comment by 0xKT** [comment]
大家有什么想法？或者其他的解决方法，可以提一下。

**Comment by leeduckgo** [comment]
没错，解决方案和你说的一致，部署所有支持链的合约，这个在前端进行配置就好了。
